### PR TITLE
Add JDK_NATIVE_OPTIONS to all openjdk test targets

### PR DESCRIPTION
**Pull request by yanshan24**
Closes: #2398 
Signed-off-by: Yanshan Wan <yanshan@ualberta.ca>

### CODE REVIEW

**Comment by smlambert** [comment]
run aqa --sdk_resource nightly --build_list openjdk --target sanity.openjdk --jdk_version 11 --jdk_impl hotspot --platform x86-64_linux

**Review by smlambert — APPROVED** [review]

**Review by karianna — APPROVED** [review]
lgtm

**Comment by sophia-guo** [comment]
run aqa --sdk_resource nightly --build_list openjdk --target -jdk_custom --jdk_version 11 --jdk_impl hotspot --platform x86-64_linux

**Comment by sophia-guo** [comment]
run aqa --sdk_resource nightly --build_list openjdk --target jdk_custom --jdk_version 11 --jdk_impl hotspot --platform x86-64_linux

**Comment by llxia** [comment]
Is there a reason we do not add `JDK_NATIVE_OPTIONS` into `JTREG_BASIC_OPTIONS`? In this way, we do not need to modify all the targets in the playlist.

**Comment by sophia-guo** [comment]
We probably need to pay attention to:

1. `-nativepath:` option can't be duplicate. That is `$(JDK_NATIVE_OPTIONS) $(JVM_NATIVE_OPTIONS)` won't work. Also `-nativepath:` option doesn't support comma or semi comma separated values.
2. Component jdk and hotspot has different nativepath.

So we will need to set a rule that add `$(JDK_NATIVE_OPTIONS)` to all jdk test targets and `$(JVM_NATIVE_OPTIONS)` to all  hotspot targets.

**Comment by sophia-guo** [comment]
@yanshan24 as talked in the meeting we can add the JDK_NATIVE_OPTIONS to all <testCaseName> started by _jdk__ except _jdk_custom_. I double checked there are actually two more testcases _<testCaseName>build</testCaseName>_ and _<testCaseName>jfc_demo</testCaseName>_ need to add JDK_NATIVE_OPTIONS. To avoid confusion could you also update these two testCaseName with prefixed jdk_, e.g  _<testCaseName>jdk_build</testCaseName>_ and _<testCaseName>jdk_jfc_demo</testCaseName>_ ? Thanks.

**Comment by yanshan24** [comment]
> @yanshan24 as talked in the meeting we can add the JDK_NATIVE_OPTIONS to all started by _jdk__ except _jdk_custom_. I double checked there are actually two more testcases _build_ and _jfc_demo_ need to add JDK_NATIVE_OPTIONS. To avoid confusion could you also update these two testCaseName with prefixed jdk_, e.g _jdk_build_ and _jdk_jfc_demo_ ? Thanks.

@sophia-guo For hotspot targets, do I need to add `$(JVM_NATIVE_OPTIONS)` or I should leave them unchanged?

**Comment by sophia-guo** [comment]
> For hotspot targets, do I need to add $(JVM_NATIVE_OPTIONS) or I should leave them unchanged?

Yes, for hotspot targets if it's without $(JVM_NATIVE_OPTIONS) please add it. That's  good point. Thanks @yanshan24

**Comment by smlambert** [comment]
@yanshan24 - this is approved, but I will wait to merge it until you push the JVM_NATIVE_OPTIONS change that you and Sophia discussed.

**Comment by yanshan24** [comment]
@sophia-guo @smlambert for tests start with `<testCaseName>jdk11_` such as `<testCaseName>jdk11_tier1_pack200</testCaseName>`, are they considered jdk test targets?

**Comment by smlambert** [comment]
re: https://github.com/AdoptOpenJDK/openjdk-tests/pull/2414#issuecomment-811600276, yes, consider the jdk11* targets to also be jdk targets that could be updated.

**Comment by yanshan24** [comment]
> re: [#2414 (comment)](https://github.com/AdoptOpenJDK/openjdk-tests/pull/2414#issuecomment-811600276), yes, consider the jdk11* targets to also be jdk targets that could be updated.

Thanks @smlambert . I have pushed the `$(JVM_NATIVE_OPTIONS)` change.

**Comment by smlambert** [comment]
Grinder for jdk_net/jdk11/openj9: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/8111

File: openjdk/playlist.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
 			<variation>Mode1000</variation>
 		</variations>
 		<command>$(JAVA_COMMAND) -Xmx512m -jar $(Q)$(TEST_RESROOT)$(D)jtreg$(D)lib$(D)jtreg.jar$(Q) \
-	$(JTREG_BASIC_OPTIONS) $(CUSTOM_NATIVE_OPTIONS) -vmoptions:$(Q)-Xmx512m $(JVM_OPTIONS)$(Q) \
+	$(JTREG_BASIC_OPTIONS) $(JVM_NATIVE_OPTIONS) $(CUSTOM_NATIVE_OPTIONS) -vmoptions:$(Q)-Xmx512m $(JVM_OPTIONS)$(Q) \
```
Comment (sophia-guo) [inline openjdk/playlist.xml:25]: jdk_custom has to figure out its own `$(CUSTOM_NATIVE_OPTIONS)` so we can leave jdk_cusotm native option as is.

**Review by sophia-guo — COMMENTED** [review]
Leave jdk_custom as is (e.g. remove $(JVM_NATIVE_OPTIONS) from jdk_custom options) the PR should be good to go.

**Review by sophia-guo — APPROVED** [review]

**Comment by Icohedron** [comment]
run aqa --sdk_resource nightly --build_list openjdk --target sanity.openjdk --jdk_version 11 --jdk_impl hotspot --platform x86-64_linux

**Pull request merged**